src/AggregateFunctions/AggregateFunctionFactory.cpp

```diff
@@ -242,13 +255,21 @@ AggregateFunctionPtr AggregateFunctionFactory::getImpl(
                 combinator_name);
         }
 
-        DataTypes nested_types = combinator->transformArguments(argument_types);
-        Array nested_parameters = combinator->transformParameters(parameters);
+        size_t combinator_parameters_number = combinator->getNumberOfParameters();
+        if (parameters.size() < combinator_parameters_number)
+            throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+                    "Incorrect number of parameters for aggregate with '{}' combinator: expected {}, got {}",
+                    combinator->getName(), combinator_parameters_number, parameters.size());
 
+        Array nested_parameters(parameters.begin(), parameters.end() - combinator_parameters_number);
```

Comment:
The second argument should be `parameters.begin() + combinator_parameters_number`, shouldn't it?

Reply:
We use first parameters for nested function and last `combinator_parameters_number` for combinator